### DISCUSSION TITLE
print-schema > "stream did not contain valid UTF8"

### DISCUSSION THREAD

**Discussion by bocksdin** [diesel-rs/diesel#3346] [Q&A] [answered]
Using diesel@2.0.0, I ran `diesel print-schema > src/schema.rs`. This generated a schema.rs file with contents I'd expected. However, when trying to reference the file in a mod statement, the linter tells me "couldn't read src\schema.rs: stream did not contain valid UTF-8." Same issue when using `diesel_ext > src/models.rs`. Has anyone encountered this issue before?

**Comment by weiznich** [comment, marked as answer]
That's hard to answer. Can you provide a reproducible answer for this? Or at least share your schema.rs file.

**Reply by bocksdin** [reply]
The file can be viewed [here](https://drive.google.com/file/d/1-fYDqaW4b-tjwhj75fiECKzwhRw0oxll/view?usp=sharing), but you'll have to download it as Google Drive thinks it's an audio file.
I'm using the schema from an external, already populated database. The schema is not coming from the migrations folder.

1. cargo install diesel_cli --no-default-features --features postgres
2. .env file with DATABASE_URL
3. diesel setup
4. diesel print-schema > src/schema.rs
5. (In main.rs) mod schema; -> error

FWIW:
Using Windows 10 19044.2006
DB hosted on bit.io

**Reply by weiznich** [reply]
The schema file reports being `UTF-16` encoded. I assume that is caused while piping using your shell. You should be able to fix this by using your favored editor, change the file encoding there and save the file again. 
```
src/schema.rs: Unicode text, UTF-16, little-endian text, with CRLF line terminators
```

Likely that's something we cannot fix directly in diesel as that's caused probably caused by the pipe operator of your command line.

**Reply by bocksdin** [reply]
Ah, that makes sense. Okay, I'll figure that out. Thanks.

**Reply by bocksdin** [reply]
That did the trick! Thank you!
